### PR TITLE
fix: CI Artifact Names, Android and Firebase Conf. Injection

### PR DESCRIPTION
**Pull request by Soap-141**
GitHub Issue: #

## Proposed Changes

 - [x] Bug fix
 - [ ] Feature
 - [ ] Code style update (formatting)
 - [ ] Refactoring (no functional changes, no api changes)
 - [x] Build or CI related changes
 - [ ] Documentation content changes
 - [ ] Other, please describe:

## Description

- Fixes an issue with the artifacts.
- Fixes an issue with Android CI/CD.
- Adds CI/CD Firebase Configuration injection.

## Impact on version

- [ ] **Major**
  - The template structure was changed.
- [ ] **Minor**
  - New functionalities were added.
- [x] **Patch**
  - A bug in behavior was fixed.
  - Documentation was changed.

### Always applicable
No matter your changes, these checks always apply.
- [x] Your conventional commits are aligned with the **Impact on version** section.
- [x] Updated [CHANGELOG.md](../CHANGELOG.md).
  - Use the latest Major.Minor.X header if you do a **Patch** change.
  - Create a new Major.Minor.X header if you do a **Minor** or **Major** change.
  - If you create a new header, it aligns with the **Impact on version** section and matches what is generated in the build pipeline.
- [x] Documentation files were updated according with the changes.
  - Update `README.md` and `src/cli/CLI.md` if you made changes to **templating**.
  - Update `AzurePipelines.md` and `src/app/README.md` if you made changes to **pipelines**.
  - Update `Diagnostics.md` if you made changes to **diagnostic tools**.
  - Update `Architecture.md` and its diagrams if you made **architecture decisions** or if you introduced new **recipes**.
  - ...and so forth: Make sure you update the documentation files associated to the recipes you changed. Review the topics by looking at the content of the `doc/` folder.
- [x] Images about the template are referenced from the [wiki](https://github.com/nventive/FlutterApplicationTemplate/wiki/Images) and added as images in this git.
- [x] TODO comments are hints for next steps for users of the template and not planned work.

### Contextual
Based on your changes these checks may not apply.
- [x] Automated tests for the changes have been added/updated.
- [x] Tested on all relevant platforms

### CODE REVIEW

File: build/steps-build-android.yml [hunk 1]
```diff
@@ -31,6 +31,11 @@ steps:
 
 - template: templates/flutter-install.yml
 
+- template : templates/replace-firebase-config.yml
+  parameters:
+    pathToSrc: '${{ parameters.pathToSrc }}'
+    isP
```
Comment (Soap-141) [inline build/steps-build-android.yml:37]: @Marc-Antoine-Soucy I think there is a typo here, is it normal?

Comment (Marc-Antoine-Soucy) [reply build/steps-build-android.yml:37]: I don't think so

Comment (Marc-Antoine-Soucy) [reply build/steps-build-android.yml:37]: It just looks like temporary blindness to me on my part

Comment (Soap-141) [reply build/steps-build-android.yml:37]: Ok I'll fix it.

**Review by takla21 — APPROVED** [review]

File: build/azure-pipelines.yml [hunk 2]
```diff
@@ -100,7 +108,7 @@ stages:
 
 - stage: Build_Production
   dependsOn: Build_Staging
-  condition: and(succeeded(), eq(variables['IsLightBuild'], 'false'))
+  condition: and(succeeded(), eq(variables['IsPullRequestBuild'], 'false'))
```
Comment (Marc-Antoine-Soucy) [inline build/azure-pipelines.yml:111]: dont we want to use the internalfirebaseOptions when we make non-prod internal builds?

Comment (Soap-141) [reply build/azure-pipelines.yml:111]: The comment is probably misaligned, but I'm sure I'm using internal for staging.

Comment (Soap-141) [reply build/azure-pipelines.yml:111]: image

Comment (Marc-Antoine-Soucy) [reply build/azure-pipelines.yml:111]: Oh you're right

**Review by Marc-Antoine-Soucy — APPROVED** [review]

**Pull request merged**